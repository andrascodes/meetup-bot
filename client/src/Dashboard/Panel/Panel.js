import React from 'react'

export const Panel = props => {
  const { previous, number } = props
  
  let panelTheme = `panel-info`
  if(number > previous) {
    panelTheme = `panel-success`
  }
  else if(number === previous) {
    panelTheme = `panel-warning`
  }
  else if(number < previous) {
    panelTheme = `panel-danger`
  }

  return (
    <div className={`panel ${panelTheme}`}>
      <div className="panel-heading">
        <div className="row">
          <div className="col-xs-3">
            <i className={`fa ${props.icon} fa-5x`}></i>
          </div>
          <div className="col-xs-9 text-right">
            <div className="huge" style={{ fontSize: `${40}px` }}>{props.number}</div>
            <div>{props.label}</div>
          </div>
        </div>
      </div>
    </div>
  )
}

Panel.PropTypes = {
  icon: React.PropTypes.string,
  number: React.PropTypes.number.isRequired,
  previous: React.PropTypes.number,
  label: React.PropTypes.string.isRequired
}