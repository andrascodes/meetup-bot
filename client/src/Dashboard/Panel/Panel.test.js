import React from 'react'
import renderer from 'react-test-renderer'

import { Panel } from './Panel'

describe('Panel', () => {

  test('renders with panel-info class when previous is not set', () => {
    const tree = renderer.create(
      <Panel
        label="Panel"
        icon="fa-car"
        number="1"
      />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('renders with panel-warning class when previous equals number', () => {
    const tree = renderer.create(
      <Panel
        label="Panel"
        icon="fa-car"
        number="1"
        previous="1"
      />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
  
  test('renders with panel-success class when previous is less than number', () => {
    const tree = renderer.create(
      <Panel
        label="Panel"
        icon="fa-car"
        number="1"
        previous="0"
      />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
  
  test('renders with panel-danger class when previous is more than number', () => {
    const tree = renderer.create(
      <Panel
        label="Panel"
        icon="fa-car"
        number="1"
        previous="2"
      />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

})