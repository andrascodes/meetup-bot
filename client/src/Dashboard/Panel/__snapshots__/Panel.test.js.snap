exports[`Panel renders with panel-danger class when previous is more than number 1`] = `
<div
  className="panel panel-danger">
  <div
    className="panel-heading">
    <div
      className="row">
      <div
        className="col-xs-3">
        <i
          className="fa fa-car fa-5x" />
      </div>
      <div
        className="col-xs-9 text-right">
        <div
          className="huge"
          style={
            Object {
              "fontSize": "40px",
            }
          }>
          1
        </div>
        <div>
          Panel
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Panel renders with panel-info class when previous is not set 1`] = `
<div
  className="panel panel-info">
  <div
    className="panel-heading">
    <div
      className="row">
      <div
        className="col-xs-3">
        <i
          className="fa fa-car fa-5x" />
      </div>
      <div
        className="col-xs-9 text-right">
        <div
          className="huge"
          style={
            Object {
              "fontSize": "40px",
            }
          }>
          1
        </div>
        <div>
          Panel
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Panel renders with panel-success class when previous is less than number 1`] = `
<div
  className="panel panel-success">
  <div
    className="panel-heading">
    <div
      className="row">
      <div
        className="col-xs-3">
        <i
          className="fa fa-car fa-5x" />
      </div>
      <div
        className="col-xs-9 text-right">
        <div
          className="huge"
          style={
            Object {
              "fontSize": "40px",
            }
          }>
          1
        </div>
        <div>
          Panel
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Panel renders with panel-warning class when previous equals number 1`] = `
<div
  className="panel panel-warning">
  <div
    className="panel-heading">
    <div
      className="row">
      <div
        className="col-xs-3">
        <i
          className="fa fa-car fa-5x" />
      </div>
      <div
        className="col-xs-9 text-right">
        <div
          className="huge"
          style={
            Object {
              "fontSize": "40px",
            }
          }>
          1
        </div>
        <div>
          Panel
        </div>
      </div>
    </div>
  </div>
</div>
`;
